File: backend/src/main/java/com/discipolat/modules/testimonials/domain/Testimony.java
package com.discipolat.modules.testimonials.domain;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "testimonies")
@org.hibernate.annotations.Filter(name = "tenantFilter", condition = "tenant_id = :tenantId")
public class Testimony {

    public enum Categorie { GUERISON, DELIVRANCE, PROVISION, FAMILLE, CONVERSION, AUTRE }
    public enum Statut { BROUILLON, EN_ATTENTE, APPROUVE, REFUSE }

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private UUID tenantId;

    @Column(nullable = false)
    private String titre;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String contenu;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Categorie categorie = Categorie.AUTRE;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Statut statut = Statut.EN_ATTENTE;

    @Column(nullable = false)
    private UUID auteurId;

    private int likes = 0;
    private int commentaires = 0;

    @Column(nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    private LocalDateTime updatedAt = LocalDateTime.now();

    // Getters and setters
    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }
    public UUID getTenantId() { return tenantId; }
    public void setTenantId(UUID tenantId) { this.tenantId = tenantId; }
    public String getTitre() { return titre; }
    public void setTitre(String titre) { this.titre = titre; }
    public String getContenu() { return contenu; }
    public void setContenu(String contenu) { this.contenu = contenu; }
    public Categorie getCategorie() { return categorie; }
    public void setCategorie(Categorie categorie) { this.categorie = categorie; }
    public Statut getStatut() { return statut; }
    public void setStatut(Statut statut) { this.statut = statut; }
    public UUID getAuteurId() { return auteurId; }
    public void setAuteurId(UUID auteurId) { this.auteurId = auteurId; }
    public int getLikes() { return likes; }
    public void setLikes(int likes) { this.likes = likes; }
    public int getCommentaires() { return commentaires; }
    public void setCommentaires(int commentaires) { this.commentaires = commentaires; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }
}

File: mobile/lib/models/testimony.dart
/// Model for Testimony entity (backend TestimonyController /api/v1/testimonies).
class Testimony {
  final String id;
  final String titre;
  final String contenu;
  final String categorie;
  final String statut;
  final String auteurId;
  final int likes;
  final int commentaires;
  final DateTime createdAt;

  Testimony({
    required this.id,
    required this.titre,
    required this.contenu,
    required this.categorie,
    required this.statut,
    required this.auteurId,
    required this.likes,
    required this.commentaires,
    required this.createdAt,
  });

  factory Testimony.fromJson(Map<String, dynamic> json) => Testimony(
        id: (json['id'] ?? '').toString(),
        titre: (json['titre'] ?? json['title'] ?? '').toString(),
        contenu: (json['contenu'] ?? json['content'] ?? '').toString(),
        categorie: (json['categorie'] ?? json['category'] ?? 'AUTRE').toString(),
        statut: (json['statut'] ?? json['status'] ?? 'EN_ATTENTE').toString(),
        auteurId: (json['auteurId'] ?? json['auteur_id'] ?? '').toString(),
        likes: json['likes'] is int ? json['likes'] as int : int.tryParse('${json['likes']}') ?? 0,
        commentaires: json['commentaires'] is int ? json['commentaires'] as int : int.tryParse('${json['commentaires']}') ?? 0,
        createdAt: DateTime.tryParse(json['createdAt']?.toString() ?? '') ?? DateTime.now(),
      );
}
